services/worker/tests/job_runners/config/test_parquet_and_info.py

```diff
@@ -926,3 +931,53 @@ def long_generator() -> Iterator[Dict[str, int]]:
         builder.download_and_prepare(file_format="parquet")
         assert builder.info.dataset_size == limiter.total_bytes <= expected_max_dataset_size
         assert builder.info.splits["train"].num_examples == num_examples < expected_max_num_examples
+
+
+@pytest.mark.parametrize(
+    "validate,too_big_row_groups",
+    [
+        (None, False),
+        (ParquetFileValidator(max_row_group_byte_size=1).validate, True),
+        (ParquetFileValidator(max_row_group_byte_size=100_000).validate, False),
+    ],
+)
+def test_fill_builder_info(
+    hub_reponses_big: HubDatasetTest,
+    tmp_path: Path,
+    validate: Optional[Callable[[pq.ParquetFile], None]],
+    too_big_row_groups: bool,
+) -> None:
+    cache_dir = str(tmp_path / "test_fill_builder_info")
+    name = hub_reponses_big["name"]
+    builder = load_dataset_builder(name, cache_dir=cache_dir)
+    builder.info = datasets.info.DatasetInfo()
+    if too_big_row_groups:
+        with pytest.raises(TooBigRowGroupsError) as exc_info:
+            fill_builder_info(builder, hf_token=None, validate=validate)
+        assert isinstance(exc_info.value, TooBigRowGroupsError)
+        assert isinstance(exc_info.value.num_rows, int)
+        assert isinstance(exc_info.value.row_group_byte_size, int)
+    else:
+        fill_builder_info(builder, hf_token=None, validate=validate)
+        expected_info = hub_reponses_big["parquet_and_info_response"]["dataset_info"]
+        assert expected_info == asdict(builder.info)
+
+
+@pytest.mark.parametrize(
+    "num_rows, row_group_byte_size, max_row_group_byte_size, expected",
+    [
+        (1000, 1000, 500, 100),
+        (1000, 1000_000, 500_000, 100),
+        (123456789, 123456789, 1000, 100),
+        (987654321, 987654321, 1000, 900),
+        (1000, 10, 1000, 1000),
+        (10, 1000, 1000, 100),
+    ],
+)
+def test_get_writer_batch_size_from_row_group_size(
```

Comment:
thanks, I prefer a unit test like this than having to trust in my reviewer skills